include/boost/geometry/arithmetic/general_form.hpp

```diff
@@ -0,0 +1,161 @@
+// Boost.Geometry
+
+// Copyright (c) 2018-2019 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ARITHMETIC_GENERAL_FORM_HPP
+#define BOOST_GEOMETRY_ARITHMETIC_GENERAL_FORM_HPP
+
+#include <boost/geometry/util/math.hpp>
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/config.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace arithmetic
+{
+
+//--------------------------------------------------------------------------
+// Structure containing the General Form of a line, a*x + b*y + c == 0
+// Might be conceptized later. Therefore operations are implemented outside
+// the structure itself.
+template <typename GeneralType = double>
+struct general_form
+{
+    general_form()
+        : a(GeneralType())
+        , b(GeneralType())
+        , c(GeneralType())
+        , normalized(false)
+    {}
+
+    // Horizontal: a == 0, for example y-3=0, y==3
+    // Vertical: b == 0, for example x-2=0, x==2
+    // Through origin: c == 0
+    GeneralType a;
+    GeneralType b;
+    GeneralType c;
+    bool normalized;
+};
+
+template <typename GeneralType, typename Coordinate>
+inline
+general_form<GeneralType> construct_line(Coordinate const& x1,
+    Coordinate const& y1, Coordinate const& x2, Coordinate const& y2)
+{
+    general_form<GeneralType> result;
+    result.a = y1 - y2;
+    result.b = x2 - x1;
+    result.c = -result.a * x1 - result.b * y1;
+    return result;
+}
+
+template <typename T, typename Point>
+inline general_form<T> construct_line(Point const& a, Point const& b)
+{
+    return construct_line<T>(geometry::get<0>(a),
+                             geometry::get<1>(a),
+                             geometry::get<0>(b),
+                             geometry::get<1>(b));
+}
+
+template <typename T, typename Segment>
+inline general_form<T> construct_line(Segment const& segment)
+{
+    return construct_line<T>(geometry::get<0, 0>(segment),
+                             geometry::get<0, 1>(segment),
+                             geometry::get<1, 0>(segment),
+                             geometry::get<1, 1>(segment));
+}
+
+
+// Calculates intersection point of two infinite lines.
+// Returns true if the lines intersect.
+// Returns false if lines are parallel (or collinear, possibly opposite)
+template <typename Point, typename GeneralType>
+inline
+bool get_intersection(Point& ip,
```

Comment:
Typically then something is called `get_XXX` then it returns `XXX`. It is also a mixture of `intersects` and intersection (thought in this case it should take the point as the last output argument). But at the same time it has different arguments than these functions. I think I'd probably call it `intersection()` but put the `Point` as the last argument as it is the case with `bg::intersection()`.

Reply:
I'm OK with changing the name and putting the point as last, sorry about that.

BTW: if the infinite line will indeed be one of our concepts, the general intersection function should be able to find its intersection